examples/ota-requestor-app/linux/main.cpp

```diff
@@ -90,7 +97,7 @@ static void InitOTARequestor(void)
     // TODO: instatiate and initialize these values when QueryImageResponse tells us an image is available
     // TODO: add API for OTARequestor to pass QueryImageResponse info to the application to use for OTADownloader init
     OTAImageProcessorParams ipParams;
-    ipParams.imageFile = CharSpan("test.txt");
+    ipParams.imageFile = CharSpan(gOtaDownloadPath);
```

Comment:
This is wrong.  It will include trailing garbage in the span, no?  Was this tested?

I woul think this should be using `CharSpan::fromCharString`.

Reply:
You are correct. The reason it works now is because all usages of the imageFile charspan in the OTA code simply use it as a null-terminated pointer (i.e. imageFile.data()). So as long as its null terminated, everything works.
However, the data span thinks the size is 256, instead of the actual size.

Reply:
> The reason it works is because all usages of the imageFile charspan in the OTA code simply use it as a null-terminated pointer 

That seems like a bug, no?  Nothing requires a CharSpan to be null-terminated.

Reply:
Indeed- I will file a ticket for this.
At the platform layer, this imageFile is passed to C-like APIs which expect it to be a NULL-terminated C string. So, I think using a "std::string_view" like container or a simple char* might make more sense thank a Char/ByteSpan, IMO.